tests/test_subscriptions.py

```diff
@@ -162,6 +163,7 @@ def test_get(self):
         self.assertEqual(self.endpoint, Endpoint.from_subscription(json_response))
         self.assertEquals(self.hmac_key_id, json_response['hmac_key_id'])
         self.assertNotIn('hmac_secret_key', json_response)
+        self._delete_subscription(uuid=json_response['uuid'])
```

Comment:
Please use a `try: finally:` block. This will allow the subscription to be deleted even if the assertions fail.

This should be applied throughout this PR.

Reply:
[addcleanup](https://docs.python.org/3/library/unittest.html#unittest.TestCase.addCleanup) would be the preferred functions, and it should be used close to the beginning of the test.

Reply:
added

Reply:
They functionally accomplish the same thing. It's your preference.

Reply:
reverted to try:finally: need verification of deletion.